### DISCUSSION TITLE
Support Cloudflare Queues?

### DISCUSSION THREAD

**Discussion by switz** [bbc/sqs-consumer#479] [Feature Request]
### Background

Cloudflare just released support for http pull in their queues. Would it be in-scope to this library to support an alternative form of sqs?

https://developers.cloudflare.com/queues/reference/pull-consumers/
https://developers.cloudflare.com/queues/platform/changelog/#support-for-pull-based-consumers

### Objectives

1. Support Cloudflare queue polling with a similar API

**Comment by nicholasgriffintn** [comment]
You had me intrigued! I use CloudFlare personally, so this is definitely interesting to me on a personal level, however, the BBC does not currently, so wouldn't be something that we would build in this package.

Alongside that, it is a different way of working to SQS as it uses an API that doesn't have as much functionality and doesn't work with the SQS SDK, one to one. Given the name, this library can only really ever be for AWS SQS or a similarly compliant tool.

That all said, I am intrigued, I've started something off here (completely untested at the time of writing this comment):

https://github.com/bbc/cloudflare-queue-consumer/blob/main/src/consumer.ts

It seems like a fairly simple thing to do that we could get a package together for.

**Reply by switz** [reply]
sick, I'll move my attention over there and contribute as I can! thanks so much.
